pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -66,15 +73,27 @@ func (f *Factory) CreateIfMissing(ctx context.Context, datacenter string, machin
 	if err != nil {
 		return fmt.Errorf("checking for template: %v", err)
 	}
+	osFamily := machineConfig.Spec.OSFamily
+	resizeBRDisk := false
+
 	if err == nil && len(templateFullPath) > 0 {
 		machineConfig.Spec.Template = templateFullPath // TODO: move this out of the factory into the defaulter, it's a side effect
 		logger.V(2).Info("Template already exists. Skipping creation", "template", machineConfig.Spec.Template)
+
+		templateName := filepath.Base(templateFullPath)
+		if strings.EqualFold(string(osFamily), string(v1alpha1.Bottlerocket)) {
+			resizeBRDisk = true
+		}
+		if resizeBRDisk {
+			if err1 := f.resizeDisk(ctx, templateFullPath, templateName, f.datacenter); err1 != nil {
+				return fmt.Errorf("error resizing disk with template: %v", err1)
+			}
+		}
```

Comment:
So, I think we should not resize the disk just if the template is Bottlerocket and especially if the user had imported it. We actually have proper documentation that asks users to issue resize commands while manually importing. We should just check if the template in input has the right size. This is primarily because the user imported it, and EKS-A editing the template that user imported would not be right.

Reply:
You can probably make a function out the validate BR template disks, and re-use it.